aptos-move/aptos-vm/README.md

```diff
@@ -1,101 +1,32 @@
-   debiting the gas fee for the transaction from the submitting account's
-   balance.
-
-During execution, the runtime resolves references to code by loading the
-referenced code via the data view. One can think of this process as similar
-to linking. Then, within the context of a block of transactions---a list of
-transactions coupled with a data view---the runtime caches code and
-linked and imported modules across transactions within the block.
-The runtime tracks state changes (data updates) from one transaction
-to the next within each block of transactions; the semantics of the
-execution of a block specify that transactions are sequentially executed
-and, as a consequence, state changes of previous transactions must be
-visible to subsequent transactions within each block.
-
-## Implementation Details
-
-* The runtime top level structs are in `runtime` and `diem vm` related
-  code.
-* The transaction flow is implemented in the [`process_txn`](./src/process_txn.rs)
-  module.
-* Code caching logic and policies are defined under the [code
-  cache](../../move-vm/runtime/src/code_cache/) directory.
-* Runtime loaded code and the type system view for the runtime is defined
-  under the [loaded data](src/loaded_data/) directory.
-* The data representation of values, and logic for write set generation can
-  be found under the [value](./src/value.rs) and [data
-  cache](./src/data_cache.rs) files.
-
-## Folder Structure
-
-```
-.
-├── src                 # VM Runtime files
-│   ├── code_cache      # VM Runtime code cache
-│   ├── loaded_data     # VM Runtime loaded data types, runtime caches over code
-│   ├── unit_tests      # unit tests
-├── vm-cache-map        # abstractions for the code cache
-```
-
-## This Module Interacts With
-
-This crate is mainly used in two parts: AC and mempool use it to determine
-if it should accept a transaction or not; the Executor runs the MoveVM
-runtime to execute the program field in a SignedTransaction and convert
-it into a TransactionOutput, which contains a writeset that the
-executor need to patch to the blockchain as a side effect of this
-transaction.
+### AptosVM
+
+AptosVM is used to execute a single Aptos transaction.
+Executes both system and user transactions.
+For every user transaction, the following steps are performed:
+
+1. ***Prologue:***
+   Responsible for checking the structure of the transaction such as signature verification, balance checks on the account paying gas for this transaction, etc.
+   If prologue checks fail, transaction is discarded and there are no updates to the blockchain state.
+2. **Execution:**
+   Runs the code specified by the transaction - can be a script or an entry function.
+   Internally, the code execution is done by MoveVM which AptosVM wraps.
+   If execution is successful, produces a change set that can be (but is not yet) applied to the blockchain state.
+3. **Aborted execution (hack):**
```

Comment:
The "hack" seems out of place. It is not clear to the reader whether the current implementation is a hack that we intend to fix in some way, or something else. Probably best to explain whatever is the intention in the prose, instead of just mentioning it here.

Reply:
Removed the "hack" part.